lib/youtube-dl.js

```diff
@@ -163,25 +163,37 @@ function call(video, args1, args2, options, callback) {
  * @param {Array.<String>} data
  */
 function filterData(data) {
-  var format = data[data.length - 1].split(' - ');
+  var info = JSON.parse(data);
+  var format = info.format.split(' - ');
+  function formatDuration(duration) {
```

Comment:
this function is already available here

https://github.com/fent/node-youtube-dl/blob/master/lib/util.js#L101